### ISSUE TITLE
[Documentation]: Audit SB8 removals in the API docs

### ISSUE THREAD

**Issue by shilman** [storybookjs/storybook#25847]
### Describe the problem

I was just looking at an issue related to `mdxSource` and did a search for it in our codebase to verify that it had been removed in v8. It has been completely removed, but there is still a single reference in the docs:

https://github.com/storybookjs/storybook/blob/69fcf16720f842c12fd0eef3b9ae90ab4fbb4d61/docs/api/doc-block-canvas.md?plain=1#L231-L237

Looking around in that file, it looks like there are other v8 removals that are also still present in the docs. Somebody should do an audit and make sure that everything that we removed from code in v8 has also been removed from the docs.

**Comment by shilman** [comment]
@JReinhold pointed out that the deprecations & removals documentation request is still WIP, so hopefully that obviates this issue.

**Comment by jonniebigodes** [comment]
For context, the APIs for the Doc Blocks were already addressed, mostly during the work towards the beta by the team. The one referenced here was already removed with #25855. However, there's still a lingering reference to the storiesOf that is currently being addressed in a pull request that I've pushed, and I'm waiting on feedback. If there were any APIs for addons, that should have been addressed with the beta release, to which I asked for for a sanity check and to the best of my knowledge they were removed. If there's any lingering ones, they should have been thoroughly vetted previously.
